app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+class PostsController < ApplicationController
+  before_action :set_post, only: %i[show edit update destroy]
+  before_action :authorize, only: [:index]
+
+  def index
+    @post = Post.new
+    @posts = Post.where(user_id: current_user.id)
```

Comment:
- [ ] SUGGESTION Have into account that everything connected with data processing should be encapsulated in models and used in controllers.